### ISSUE TITLE
Different format for context object of span capture for v3 agent

### ISSUE THREAD

**Issue by toughrogrammer** [elastic/apm-agent-python#369]
I'm getting so many warning log `object mapping for [context.service] tried to parse field [service] as object, but found a concrete value`. So I start finding what data is wrong and here is my found.

My span for mysql query looks like below

```
"context": {
    "db": {
      "type": "sql",
      "statement": "SELECT %s"
    },
    "service": {
      "name": "my-service-name",
      "agent": {
        "name": "python",
        "version": "3.0.4"
      }
    }
},
```

but span for botocore is different format

```
"context": {
    "region": "ap-northeast-1",
    "operation": "PutItem",
    "service": "dynamodb"
},
```

As I use APM server 6.5.1, elasticsearch index template of apm-6.5.1 don't have concrete value field for `context.service`. I found there is no issue for urllib3 instrumenting. The different thing is `capture_span` in `Urllib3Instrumentation` pass dictionary object for `capture_span` functions third parameter but `capture_span` in `BotocoreInstrumentation` pass dictionary with concrete string value for `service` key. 

<img width="549" alt="screen shot 2018-12-24 at 11 43 01 pm" src="https://user-images.githubusercontent.com/1473538/50401922-accc2800-07d5-11e9-944f-4950b4449acf.png">

<img width="588" alt="screen shot 2018-12-24 at 11 42 53 pm" src="https://user-images.githubusercontent.com/1473538/50401924-af2e8200-07d5-11e9-80aa-8dcaf7634e08.png">

**Comment by toughrogrammer** [comment]
I'm using agent version 3.0.4 but it's same for 4.0.3.

**Comment by beniwohli** [comment]
Hi @toughrogrammer 

We don't set the `service` property in boto3/botocore spans anymore in the just-released version 4.0.3 (https://github.com/elastic/apm-agent-python/pull/367). Can you double check if the error still appears in 4.0.3?

It's possible that due to dynamic mapping, a `context.service` field with the wrong type was created. Deleting today's apm-span index in Elasticsearch could help in that case.

**Comment by toughrogrammer** [comment]
Hi @beniwohli 

There is some problems... I tried 4.0.3 but I found there is problem to transport tracing data to apm server. Below capture is our services APM transaction per minute graph.

<img width="1083" alt="screen shot 2018-12-25 at 12 38 08 am" src="https://user-images.githubusercontent.com/1473538/50402877-6f6b9880-07dd-11e9-815d-d36b1cf1db1d.png">

I rolled back to v3.0.4 at about 10 P.M., and then there was no problem. I cannot found actually what point is problem but I decided I cannot use v4.0.3. It looks like there are some problem about transporting data.

**Comment by toughrogrammer** [comment]
Maybe creating new issue for transporting data is more better.

**Comment by beniwohli** [comment]
That would be great. That graph definitely looks concerning

**Comment by toughrogrammer** [comment]
So there is no more update for v3 agent?

**Comment by beniwohli** [comment]
We will only backport security issues (should they appear) to 3.x

**Comment by beniwohli** [comment]
Closing this. The quickfix has been released as both 3.0.5 and 4.0.3
